Sprint-1/stretch/aoc-2018-day1/solution.js

```diff
@@ -0,0 +1,34 @@
+const fs = require('fs');
+const path = require('path');
+
+try {
+    const input = fs.readFileSync(path.resolve(__dirname, "./input.txt"), "utf-8")
+                    .trim()
+                    .split('\n')
+                    .map(i => parseInt(i, 10));
+
+    // Part 1: Calculate the resulting frequency
+    const part1Result = input.reduce((a, b) => a + b, 0);
+    console.log("Part 1 - Resulting Frequency:", part1Result);
+
+    // Part 2: Find the first frequency reached twice
+    let sum = 0;
+    let set = new Set([0]);
+    let length = input.length;
+    let part2Result = null;
+
+    for (let i = 0; ; i = (i + 1) % length) {
```

Comment:
How does one get Part 2 spec? I only saw Part 1 spec.

Why not use the "typical form" of for loop, `for (let i = 0; i < length; i++) {` ?

Reply:
Part 2 is to find the first repeated frequency which loops through the numbers and adding up to a running total, if the total has been encountered before then the loop ends. The loop ensures the program keeps running until a repeated frequency is found

Reply:
The reason for the infinite for loop using a modulo is because the requirement is to repeat the input list until finding a repeating frequency. A normal for loop would only iterate over the list once and in some cases you would need to loop back to the beginning of the list to continue summing up the frequencies. Using the modulo operation (i = (i + 1) % length) allows us to wrap around the input list, repeating it without having to manually reset the loop or count the iterations.